sys/parsoid.js

```diff
@@ -230,18 +236,28 @@ class ParsoidService {
         return new URI(path);
     }
 
-    getFallbackBucketURI(rp, format, tid) {
-        return this.getOldStashBucketURI(rp, format, tid);
-    }
-
     _getContentWithFallback(hyper, rp, format, tid) {
         return hyper.get({
-            uri: this.getNGBucketURI(rp, format, tid)
+            uri: this.getOldLatestBucketURI(rp, format, tid)
+        })
+        .catch({ status: 404 }, (e) => {
+            if (rp.revision && tid) {
+                return hyper.get({
+                    uri: this.getStashBucketURI(rp.domain, rp.title, rp.revision, tid)
+                })
+                .then((res) => Object.assign(
+                    { status: 200 },
+                    JSON.parse(res.body.toString('utf8'))[format])
```

Comment:
It wouldn't be a bad idea to surround this by a try/catch perhaps?

Reply:
we are already inside a promise, what would we achieve with a try-catch?

Reply:
A nice error msg. But this was just an observation, we can include it later